x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -667,6 +667,10 @@ Collection<Object> createComponents(
             return Collections.singletonList(new SecurityUsageServices(null, null, null, null, null, null));
         }
 
+        // The settings in `environment` may have additional values over what was provided during construction
+        // See Plugin#additionalSettings()
+        this.settings = environment.settings();
```

Comment:
thanks for fixing this ! Can you update the the TODO's above (in the constructor) to with a comment that warns that the the setting read in the constructor will not honor additionalSettings and that the value is update here.